sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -137,13 +139,79 @@ public AnalyzedRelation analyze(Node node, Analysis analysis) {
     }
 
     @Override
-    protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext context) {
-        return process(node.getQueryBody(), context);
+    protected AnalyzedRelation visitQuery(Query node, StatementAnalysisContext statementContext) {
+        // In case of Set Operation or VALUES clause
+        if (!node.getOrderBy().isEmpty() || node.getLimit().isPresent() || node.getOffset().isPresent()) {
+            QueriedRelation childRelation = (QueriedRelation) process(node.getQueryBody(), statementContext);
+
+            // Use child relation to process expressions of the "root" Query node
+            statementContext.startRelation();
+            RelationAnalysisContext relationAnalysisContext = statementContext.currentRelationContext();
+            String qualifiedName = childRelation.getQualifiedName().toString();
+            // In case of Union we need to analyze ORDER BY based on the output symbols of the left-most relation
+            if (childRelation instanceof UnionSelect) {
+                qualifiedName = ((UnionSelect) childRelation).left().getQualifiedName().toString();
```

Comment:
I don't quite undestand why this here is necessary - and I think it shouldn't be necessary.

If the childRelation ( `UnionSelect` in this case) is added, doesn't it work correctly by implementing  `getField` to use the left relation?